Jerrod_Quintana/Assignments/library.js

```diff
@@ -0,0 +1,53 @@
+var _ = {
+   map: function(arr, func) {
+     var newarr = [];
+     for (var i = 0; i < arr.length; i++){
+       newarr.push(func(arr[i]))
+     }
+     return newarr;
+
+   },
+   reduce: function(arr, func) {
+     var end = 0;
+     for (var i = 0; i < arr.length - 1; i++){
+       end += func(arr[i])
+     }
+     return end;
+
+   },
+   find: function(arr, func) {
+     var answer = [];
+     for (var i = 0; i < arr.length; i++){
+       if (func(arr[i]) == true){
+         answer.push(arr[i]);
+         return answer;
+       }
+     }
+   },
+   filter: function(arr, func) {
+     var answer = [];
+     for (var i = 0; i < arr.length; i++){
+       if (func(arr[i]) == true){
+         answer.push(arr[i]);
+       }
+     }
+     return answer
+   },
+   reject: function(arr, func) {
+     var answer = [];
+     for (var i = 0; i < arr.length; i++){
+       if (func(arr[i]) != true){
```

Comment:
You can also remove items in an array with splice rather than creating a new one.